Electron/src/Core/components/Cameras.tsx

```diff
@@ -60,9 +77,90 @@ class Cameras extends Component<IProps, IState> {
     return `http://${ip}:8080/${camera}/stream`
   }
 
+  rotate(): void {
+    let style: CSS.Properties
+    let { rotation } = this.state
+    rotation = (rotation + 90) % 360
+
+    let width = 0
+    let height = 0
+
+    for (const win of Object.keys(windows)) {
+      if (windows[win].document.getElementById(this.state.id)) {
+        width = windows[win].document.getElementById(this.state.id).clientWidth
+        height = windows[win].document.getElementById(this.state.id).clientHeight
+        break
+      }
+    }
+
+    const Xfactor = height / width
+    const Yfactor = width / height
+
+    switch (rotation) {
+      case 90:
+        style = { transform: `rotate(90deg) scaleX(${Xfactor}) scaleY(${Yfactor})` }
+        break
+      case 180:
+        style = { transform: "rotate(180deg)" }
+        break
+      case 270:
+        style = { transform: `rotate(270deg) scaleX(${Xfactor}) scaleY(${Yfactor})` }
+        break
+      default:
+        style = { transform: "" }
+        break
+    }
+
+    this.setState({ rotation, style })
+  }
+
+  saveImage(): void {
+    let camera
+    for (const win of Object.keys(windows)) {
+      if (windows[win].document.getElementById(this.state.id.toString())) {
+        camera = windows[win].document.getElementById(this.state.id.toString())
+        break
+      }
+    }
+
+    if (!camera) {
+      throw new Error("The element 'camera' wasn't found")
+    }
+
+    html2canvas(camera, {
+      scrollX: 0,
+      scrollY: -window.scrollY,
+      useCORS: true,
+      allowTaint: true,
+    })
+      .then(canvas => {
+        const imgData = canvas.toDataURL("image/png").replace("image/png", "image/octet-stream")
+        this.downloadURL(imgData)
+        return null
+      })
+      .catch(error => {
+        console.error(error)
+      })
+  }
+
+  downloadURL(imgData: string): void {
+    const filename = `./Screenshots/${new Date()
+      .toISOString()
+      // ISO string will be fromatted YYYY-MM-DDTHH:MM:SS:sssZ
+      // this regex will convert all -,T,:,Z to . (which covers to . for .csv)
+      .replaceAll(/[:\-TZ]/g, ".")}Cam${this.state.currentCamera}.png`
+
+    if (!fs.existsSync("./Screenshots")) {
+      fs.mkdirSync("./Screenshots")
+    }
+
+    const base64Image = imgData.replace("image/png", "image/octet-stream").split(";base64,").pop()
+    fs.writeFileSync(filename, base64Image!, { encoding: "base64" })
+  }
```

Comment:
Comment how this works so that it is documented somewhere, noting the changes to save to file automatically, grabbing the component based on the unique id from the screen, and mention the current issues with saving images.